payroll/models/hr_payslip_run.py

```diff
@@ -0,0 +1,60 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import date, datetime
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import fields, models
+
+
+class HrPayslipRun(models.Model):
+    _name = "hr.payslip.run"
+    _description = "Payslip Batches"
+
+    name = fields.Char(
+        required=True, readonly=True, states={"draft": [("readonly", False)]}
+    )
+    slip_ids = fields.One2many(
+        "hr.payslip",
+        "payslip_run_id",
+        string="Payslips",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    state = fields.Selection(
+        [("draft", "Draft"), ("close", "Close")],
+        string="Status",
+        index=True,
+        readonly=True,
+        copy=False,
+        default="draft",
+    )
+    date_start = fields.Date(
+        string="Date From",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        default=lambda self: fields.Date.to_string(date.today().replace(day=1)),
+    )
+    date_end = fields.Date(
+        string="Date To",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        default=lambda self: fields.Date.to_string(
```

Comment:
```suggestion
        default=lambda self: fields.Datetime.now() + relativedelta(months=+1, day=1, days=-1)
```
default can be datetime but I'm not sure function fields.Datetime.now()  include timezone.
or you can use similar date_start
```suggestion
       default=lambda self: fields.Date.today().replace(day=1) + relativedelta(months=+1, day=1, days=-1)
```